WordPress/wordpress-develop
# backport Rename Reusable blocks to Patterns

Renames the Reusable blocks to Patterns. Also adds the option to convert a block or collection of blocks to a non-synced Pattern. This then behaves as imported Patterns do in that once inserted they can be edited completely independently of any other instances of the Pattern. The synced version of the Pattern works as the existing Reusable blocks do in that editing one instance updates all instances.

From: https://github.com/WordPress/gutenberg/pull/51144



Trac ticket: https://core.trac.wordpress.org/ticket/58577

## Testing
Full testing would require GB plugin 16.1  installing.

- Before switching to PR branch add some reusable blocks
- Switch to branch and make sure the existing reusable blocks still work as expected
- Go to /wp-admin/edit.php?post_type=wp_block and check that the existing blocks have Sync Status set to Synced in the right post properties panel
- Add a new post and add a block or collection of blocks
- Select the block(s) and in the block toolbar overflow menu select the Create a pattern menu option
- The sync toggle will be off by default - leave it that way and add the new pattern
- In the block inspector search for the new pattern name and insert
- Check that inserted blocks are independent/stand-alone blocks and not a synced reusable block
- Repeat the above but toggle sync on and make sure the saved block behaves the same as the existing reusable blocks
- Go to /wp-admin/edit.php?post_type=wp_block and make sure the synced and unsynced Patterns show. Edit Patterns from this page and check that toggling the sync status in right panel works as expected


---
**This Pull Request is for code review only. Please keep all other discussion in the Trac ticket. Do not merge this Pull Request. See [GitHub Pull Requests for Code Review](https://make.wordpress.org/core/handbook/contribute/git/github-pull-requests-for-code-review/) in the Core Handbook for more details.**
